Python/Tests/DebuggerUITests/DebugProjectUITests.cs

```diff
@@ -1044,13 +1044,44 @@ class EmptyDisposable : IDisposable {
             }
         }
 
+        class AggregateDisposable : IDisposable {
```

Comment:
You can port this one: https://github.com/Microsoft/RTVS/blob/774e5c746bf0bb99d45247a4fdf082199abd2461/src/Common/Core/Impl/Disposables/DisposableBag.cs
But it's up to you.

Reply:
`VisualStudioApp` already has `OnDispose(Action)` - any reason why that won't work?

Reply:
Maybe we just need to reorder `VisualStudioApp.Dispose` to do the close all first?

Reply:
I'm just chaining the "close all" to run before the end of the existing using statement that's there to restore the global default interpreter.
```
            using (SelectDefaultInterpreter(app, interpreter))
            using (new PythonExperimentalGeneralOptionsSetter(pyService, useVsCodeDebugger: useVsCodeDebugger)) {
                var proj = OpenDebuggerProject(app);
                // do stuff
                
                // close all here before the "default interpreter" is disposed (causes interpreter change, which project subscribes to and causes reanalysis)
            }
```

Close all is already done in app.Dispose, and that's too late for me.

DisposableBag could be useful in general, but isn't saving me any code here.

Reply:
I'm not so keen on having many ways to do multiple disposals... what if we changed `SelectDefaultInterpreter` to not be disposable itself, but it becomes a method on `app` and hooks up via `OnDispose()`, and we put that after closing the project?

Given what we use `OnDispose()` for (mostly options), it would seem to make sense to reset those after stopping debugging/closing dialogs/solution anyway.